src/components/structures/RoomView.js

```diff
@@ -877,13 +877,12 @@ module.exports = React.createClass({
         // If the user is a ROU, allow them to transition to a PWLU
         if (cli && cli.isGuest()) {
             // Join this room once the user has registered and logged in
-            const signUrl = this.props.thirdPartyInvite ?
-                this.props.thirdPartyInvite.inviteSignUrl : undefined;
+            // (If we failed to peek, we may not have a valid room object.)
             dis.dispatch({
                 action: 'do_after_sync_prepared',
                 deferred_action: {
                     action: 'view_room',
-                    room_id: this.state.room.roomId,
+                    room_id: this.state.room ? this.state.room.roomId : this.state.roomId,
```

Comment:
Maybe we should always use `state.roomId`?

Reply:
Well, I wasn't sure if we keep it up to date... In `_onRoomViewStoreUpdate`, we set the new state, and it specifically mentions `this.state.roomId` may be null if it we have a room alias:

https://github.com/matrix-org/matrix-react-sdk/blob/9d3ba2b3d9a9fcc0f8ee1fc222c9da9d26ce3d25/src/components/structures/RoomView.js#L273-L283

I didn't notice any code paths that might later go back and update `this.state.roomId` after resolving an alias, so it seemed safer to check both.

@bwindels I haven't looked at this area much, so what would you recommend?

Reply:
For now, I'll add a comment about this.

Reply:
Yes, sounds good, I forgot about room alias resolving.